File: src/main/java/com/dh/ondot/schedule/core/exception/UnhandledOpenAiException.java
package com.dh.ondot.schedule.core.exception;

import com.dh.ondot.core.exception.InternalServerException;

import static com.dh.ondot.core.exception.ErrorCode.UNHANDLED_OPEN_AI;

public class UnhandledOpenAiException extends InternalServerException {
    public UnhandledOpenAiException() {
        super(UNHANDLED_OPEN_AI.getMessage());
    }

    @Override
    public String getErrorCode() {
        return UNHANDLED_OPEN_AI.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/core/exception/UnhandledOpenAiException.kt
package com.dh.ondot.schedule.core.exception

import com.dh.ondot.core.exception.ErrorCode.UNHANDLED_OPEN_AI
import com.dh.ondot.core.exception.InternalServerException

class UnhandledOpenAiException :
    InternalServerException(UNHANDLED_OPEN_AI.message) {
    override val errorCode: String get() = UNHANDLED_OPEN_AI.name
}
